File: templates/credentials/Credentials.js
/**
 * Classes de credenciais para APIs de IA
 * 
 * Este módulo fornece classes para gerenciar credenciais de diferentes
 * provedores de IA (OpenAI, Anthropic, OpenRouter, etc.)
 */

/**
 * Classe base abstrata para credenciais de API
 */
class Credentials {
    /**
     * Obtém a chave da API
     * 
     * @returns {string} Chave da API ou string vazia se não configurada
     */
    getApiKey() {
        throw new Error('Método getApiKey() deve ser implementado');
    }
}

/**
 * Credenciais para OpenAI
 */
class OpenAICredentials extends Credentials {
    /**
     * Construtor
     * 
     * @param {string|null} apiKey - Chave da API (opcional, usa variável de ambiente se não fornecida)
     */
    constructor(apiKey = null) {
        super();
        this._apiKey = apiKey;
    }
    
    /**
     * Obtém a chave da API OpenAI
     * 
     * @returns {string} Chave da API ou string vazia se não configurada
     */
    getApiKey() {
        if (this._apiKey) {
            return this._apiKey;
        }
        return process.env.OPENAI_API_KEY || '';
    }
}

/**
 * Credenciais para Anthropic (Claude)
 */
class AnthropicCredentials extends Credentials {
    /**
     * Construtor
     * 
     * @param {string|null} apiKey - Chave da API (opcional, usa variável de ambiente se não fornecida)
     */
    constructor(apiKey = null) {
        super();
        this._apiKey = apiKey;
    }
    
    /**
     * Obtém a chave da API Anthropic
     * 
     * @returns {string} Chave da API ou string vazia se não configurada
     */
    getApiKey() {
        if (this._apiKey) {
            return this._apiKey;
        }
        return process.env.ANTHROPIC_API_KEY || '';
    }
}

/**
 * Credenciais para OpenRouter
 */
class OpenRouterCredentials extends Credentials {
    /**
     * Construtor
     * 
     * @param {string|null} apiKey - Chave da API (opcional, usa variável de ambiente se não fornecida)
     */
    constructor(apiKey = null) {
        super();
        this._apiKey = apiKey;
    }
    
    /**
     * Obtém a chave da API OpenRouter
     * 
     * @returns {string} Chave da API ou string vazia se não configurada
     */
    getApiKey() {
        if (this._apiKey) {
            return this._apiKey;
        }
        return process.env.OPENROUTER_API_KEY || '';
    }
}

module.exports = {
    Credentials,
    OpenAICredentials,
    AnthropicCredentials,
    OpenRouterCredentials
};

File: templates/credentials/Credentials.py
"""
Classes de credenciais para APIs de IA

Este módulo fornece classes para gerenciar credenciais de diferentes
provedores de IA (OpenAI, Anthropic, OpenRouter, etc.)
"""
import os
from abc import ABC, abstractmethod
from typing import Optional


class Credentials(ABC):
    """Classe base abstrata para credenciais de API"""
    
    @abstractmethod
    def get_api_key(self) -> str:
        """
        Obtém a chave da API
        
        Returns:
            Chave da API ou string vazia se não configurada
        """
        pass


class OpenAICredentials(Credentials):
    """Credenciais para OpenAI"""
    
    def __init__(self, api_key: Optional[str] = None):
        """
        Inicializa credenciais OpenAI
        
        Args:
            api_key: Chave da API (opcional, usa variável de ambiente se não fornecida)
        """
        self._api_key = api_key
    
    def get_api_key(self) -> str:
        """
        Obtém a chave da API OpenAI
        
        Returns:
            Chave da API ou string vazia se não configurada
        """
        if self._api_key:
            return self._api_key
        return os.getenv('OPENAI_API_KEY', '')


class AnthropicCredentials(Credentials):
    """Credenciais para Anthropic (Claude)"""
    
    def __init__(self, api_key: Optional[str] = None):
        """
        Inicializa credenciais Anthropic
        
        Args:
            api_key: Chave da API (opcional, usa variável de ambiente se não fornecida)
        """
        self._api_key = api_key
    
    def get_api_key(self) -> str:
        """
        Obtém a chave da API Anthropic
        
        Returns:
            Chave da API ou string vazia se não configurada
        """
        if self._api_key:
            return self._api_key
        return os.getenv('ANTHROPIC_API_KEY', '')


class OpenRouterCredentials(Credentials):
    """Credenciais para OpenRouter"""
    
    def __init__(self, api_key: Optional[str] = None):
        """
        Inicializa credenciais OpenRouter
        
        Args:
            api_key: Chave da API (opcional, usa variável de ambiente se não fornecida)
        """
        self._api_key = api_key
    
    def get_api_key(self) -> str:
        """
        Obtém a chave da API OpenRouter
        
        Returns:
            Chave da API ou string vazia se não configurada
        """
        if self._api_key:
            return self._api_key
        return os.getenv('OPENROUTER_API_KEY', '')

